pkg/cachemanager/cachemanager.go

```diff
@@ -341,7 +348,17 @@ func (c *CacheManager) manageCache(ctx context.Context) {
 
 				// stop any goroutines that were relisting before
 				// as we may no longer be interested in those gvks
+				// and wait with a timeout for the child gorountine to stop.
 				close(relistStopChan)
```

Comment:
chatting w rita offline:

add a small comment that highlights why we close the `relistStopChan` here but `waitToCloseChan` outside for the 0th iteration.